Sanity check@0.1.0

edsu7 opened the pull request:
See sanity-check.readme for tested scenarios

edsu7 committed:
[wfpm v0.8.0] added starting template for sanity-check@0.1.0

edsu7 committed:
initial commit

edsu7 committed:
updated main.py

- changed query behaviour to identify empty/false projects

lindaxiang commented on sanity-check/README.md:
```diff
@@ -0,0 +1,69 @@
+# Package differentiate-json
+
+Updates provided `file_info_tsv` with additional fields from Clinical. This step is to avoid mismatching info for the same field between Clinical.SONG and Submission.SONG.
+
```
`experiment_info_tsv`?

edsu7 replied:
Updated

lindaxiang commented on sanity-check/README.md:
```diff
@@ -0,0 +1,69 @@
+
+## Inputs
+
+`File_info_tsv` - See argo-data-submission-wf
```
`experiment_info_tsv`

edsu7 replied:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/README.md?plain=1#L7

lindaxiang commented on sanity-check/README.md:
```diff
@@ -0,0 +1,69 @@
+# Package differentiate-json
```
Package sanity-check?

edsu7 replied:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/README.md?plain=1#L1

lindaxiang commented on sanity-check/README.md:
````diff
@@ -0,0 +1,69 @@
+params file(s) can be found in the `tests` folder.
+
+```
+nextflow run icgc-argo/argo-data-submission/differentiate-json/main.nf -r differentiate-json.v0.1.0 -params-file <your-params-json-file>
````
differentiate-json.v0.1.0?

edsu7 replied:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/README.md?plain=1#L59

lindaxiang commented on sanity-check/main.nf:
```diff
@@ -0,0 +1,94 @@
+params.clinical_url="https://clinical.qa.argo.cancercollaboratory.org"
+params.api_token=""
+params.experiment_info_tsv="NO_FILE1"
+params.skip_sanity_check="NO_FILE2"
```
skip_sanity_check: use `boolean` instead of `NO_FILE`? Is`skip_dup_check` more proper?

edsu7 replied:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/main.nf#L50

lindaxiang commented on sanity-check/main.py:
```diff
@@ -0,0 +1,214 @@
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-x', '--experiment_info_tsv', dest='experiment_info_tsv', type=str,required=True,
+                        help='TSV contain experiment info and submitter IDs')
+    parser.add_argument('-f', '--force', action='store_true',help='JSON file containing metadata payload')
```
`JSON file containing metadata payload`?

edsu7 replied:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/main.py#L50

lindaxiang commented on sanity-check/main.py:
```diff
@@ -0,0 +1,214 @@
+    parser.add_argument('-t', '--api_token', dest='api_token', type=str,
+                        help='Platform UUID token', required=True)
+    parser.add_argument('-c', '--clinical_url', dest='clinical_url', type=str,
+                        help='Clinical SONG API interaction URL', required=True)
```
Clinical API interaction URL

edsu7 replied:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/main.py#L44-L45

lindaxiang commented on sanity-check/main.py:
```diff
@@ -0,0 +1,214 @@
+            clinical_metadata
+        )
+        if not args.force:
+            check_study_exists(
```
`check_study_exists` should be always checked.

edsu7 replied:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/main.py#L66-L70

lindaxiang commented on sanity-check/main.py:
```diff
@@ -0,0 +1,214 @@
+        ### Handles instances where sample has existing analyses
+            for analysis in response.json():
+                ### If analysis is suppressed; we ignore
+                if analysis["analysisState"]=="PUBLISHED" and analysis["experiment"]["experimental_strategy"]==metadata.get('experimental_strategy'):
```
Need to add `analysisType=="sequencing_experiment"`

edsu7 replied:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/main.py#L190-L194

lindaxiang commented on sanity-check/main.py:
```diff
@@ -0,0 +1,214 @@
+    response=requests.get(endpoint,headers=headers)
+    
+    if response.status_code==404:
+        sys.exit("Program %s does not exist. Please verify program code is correct. Otherwise contact DCC-admin for help to troubleshoot." % (metadata.get('program_id')))
```
Program %s does not exist in Song.

edsu7 replied:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/main.py#L168

lindaxiang commented on sanity-check/README.md:
```diff
@@ -0,0 +1,69 @@
+#### Successful instance
+
+An `updated_file_info.tsv` is generated
+
```
updated_experiment_info.tsv

edsu7 replied:
Updated

edsu7 committed:
Updates to address typos and help statements

- cleaned up `README.md`
- fixed typos in `main.py`
- converted `skip-sanity-check` argument to `skip-duplicate-check`. Also now boolean

lindaxiang reviewed (approved):
LGTM
